### PR TITLE
MG-888 - Add certs API tests

### PR DESCRIPTION
**Pull request by felixgateru**
# What type of PR is this?
This is a feature because it adds api tests for the certs service.

## What does this do?
This pr introduces `/certs/api/endpoint_test.go` file for api tests.

## Which issue(s) does this PR fix/relate to?

- Related Issue #888 

## Have you included tests for your changes?
Yes, the test file is the change.

## Did you document any new/modified feature?
No,

### CODE REVIEW

File: certs/api/endpoint_test.go [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+	for _, tc := range cases {
+		req := testRequest{
+			client:      cs.Client(),
+			method:      http.MethodPost,
+			url:         fmt.Sprintf("%s/certs", cs.URL),
+			contentType: tc.contentType,
+			token:       tc.token,
+			body:        strings.NewReader(tc.request),
+		}
+		svcCall := svc.On("IssueCert", mock.Anything, tc.token, tc.thingID, tc.ttl).Return(cert, tc.svcErr)
```
Comment (rodneyosodo) [inline certs/api/endpoint_test.go]:
```suggestion
		svcCall := svc.On("IssueCert", mock.Anything, tc.token, tc.thingID, tc.ttl).Return(tc.svcRes, tc.svcErr)
```

File: certs/api/endpoint_test.go [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+			desc:     "view with empty token",
+			token:    "",
+			serialID: serial,
+			status:   http.StatusUnauthorized,
+			svcRes:   certs.Cert{},
+			svcErr:   nil,
+			err:      apiutil.ErrBearerToken,
+		},
+		{
+			desc:     " view no existing cert",
```
Comment (rodneyosodo) [inline certs/api/endpoint_test.go]:
```suggestion
			desc:     "view no existing cert",
```

File: certs/api/endpoint_test.go [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+			offset:  0,
+			limit:   10,
+			query:   "",
+			status:  http.StatusUnauthorized,
+			svcRes:  certs.Page{},
+			svcErr:  nil,
+			err:     apiutil.ErrBearerToken,
+		},
+		{
+			desc:    "list with limit exceeding",
```
Comment (rodneyosodo) [inline certs/api/endpoint_test.go]:
```suggestion
			desc:    "list with limit exceeding max limit",
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: certs/api/endpoint_test.go [hunk 4]
```diff
@@ -0,0 +1,536 @@
...
+	if err != nil {
+		return nil, err
+	}
+	if tr.token != "" {
+		req.Header.Set("Authorization", apiutil.BearerPrefix+tr.token)
+	}
+	if tr.contentType != "" {
+		req.Header.Set("Content-Type", tr.contentType)
+	}
+	req.Header.Set("Referer", tr.referer)
```
Comment (rodneyosodo) [inline certs/api/endpoint_test.go]: We don't need this

File: certs/api/endpoint_test.go [hunk 5]
```diff
@@ -0,0 +1,536 @@
...
+
+	mux := httpapi.MakeHandler(svc, logger, "")
+	return httptest.NewServer(mux), svc
+}
+
+func TestIssueCert(t *testing.T) {
+	cs, svc := newCertServer()
+	defer cs.Close()
+
+	validReqString := `{
```
Comment (rodneyosodo) [inline certs/api/endpoint_test.go]: make this inline

```go
validReqString := `{"thing_id": "%s","ttl": "%s"}`
invalidReqString := `{"thing_id": "%s","ttl": %s}`
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by rodneyosodo — APPROVED** [review]

**Review by arvindh123 — APPROVED** [review]

**Review by JeffMboya — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**